lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -348,33 +348,56 @@ private function addMissingItemsFromOrderByToSelect(SelectStatement $AST)
     /**
      * Generates new SQL for statements with an order by clause
      *
-     * @param array           $sqlIdentifier
-     * @param string          $innerSql
-     * @param string          $sql
-     * @param OrderByClause   $orderByClause
+     * @param array $sqlIdentifier
+     * @param string $innerSql
+     * @param string $sql
+     * @param OrderByClause $orderByClause
      *
      * @return string
      */
-    private function preserveSqlOrdering(array $sqlIdentifier, $innerSql, $sql, $orderByClause)
-    {
+    private function preserveSqlOrdering(array $sqlIdentifier, $innerSql, $sql, $orderByClause) {
         // If the sql statement has an order by clause, we need to wrap it in a new select distinct
         // statement
-        if (! $orderByClause instanceof OrderByClause) {
+        if (!$orderByClause instanceof OrderByClause) {
             return $sql;
         }
 
         // Rebuild the order by clause to work in the scope of the new select statement
         /* @var array $orderBy an array of rebuilt order by items */
         $orderBy = $this->rebuildOrderByClauseForOuterScope($orderByClause);
+        $orderByFields = str_replace([' DESC', ' ASC'], ['', ''], $orderBy);
```

Comment:
You may have something like `ORDER BY ASCII ASC` - that would be replaced with `ORDER BYII` here - no-go.